### DISCUSSION TITLE
listening for each throw in the process

### DISCUSSION THREAD

**Discussion by tareksalem** [nodejs/help#4939] [Q&A]
### What is the problem this feature will solve?

Currently, we have process listeners to listen for `uncaughtException`, `unhandledRejection`, `rejectionHandled` to catch the unhandled errors in sync and async but we need another event to be emitted each time we have a `throw` or `.reject` statements in the process even if it is handled.

### What is the feature you are proposing to solve the problem?

suppose we have a function that throws an error without a catch and this function is being called in another one that has caught, in this case, no event will be emitted however for logging purposes sometimes we need to log each error that happens in single-function one by one a

**Comment by mscdex** [comment]
This seems like something that should be handled in your code, not in node core. Alternatively you could write something that would transpile your code to automatically add such logging within `catch` scopes.

**Comment by tareksalem** [comment]
> 

I think this is doable via a transpiler however nodejs is a runtime and targets to work with raw js without transpilers so supporting something like this would be better as nodejs has the management of the process not the transpiler.

what about if I created a PR regarding to this, do you thing it's a good idea?

**Comment by bnoordhuis** [comment]
No changes are required because you can already do it today through the `Debugger.setPauseOnExceptions` inspector command with `{ state: "all" }`, then waiting for the `Debugger.paused` events to come in.

See https://nodejs.org/docs/latest/api/inspector.html for more details. I'll convert this to a discussion.

**Reply by tareksalem** [reply]
Thanks for your reply

Does this work for production?

I need to stream all the errors that happen in multiple code layers dynamically

**Reply by bnoordhuis** [reply]
Sure. There's a performance impact you'll need to evaluate but TANSTAAFL, of course.

**Reply by tareksalem** [reply]
why this is not implemented in process listener for better implementation?

**Reply by bnoordhuis** [reply]
I think what you're asking is "would performance be better if it was supported in some other way by node?" The answer is "probably not."
